compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -2802,14 +2802,34 @@ abstract class GenJSCode extends plugins.PluginComponent
          *   {name1: arg1, name2: arg2, ... }
          */
 
+        def warnIfDuplicatedKey(pairs: List[(js.StringLiteral, js.Tree)]): Unit = {
+          val allKeys = pairs.collect { case (js.StringLiteral(keyName), _) => keyName }
```

Comment:
Oh. Right, I missed than I've already filtered them. It is at least correct. I'll remove the extra chech/work in a next issue.